lib/matplotlib/axis.py

```diff
@@ -1362,48 +1360,31 @@ def get_major_ticks(self, numticks=None):
         'get the tick instances; grow as necessary'
         if numticks is None:
             numticks = len(self.get_major_locator()())
-        if len(self.majorTicks) < numticks:
-            # update the new tick label properties from the old
-            for i in range(numticks - len(self.majorTicks)):
-                tick = self._get_tick(major=True)
-                self.majorTicks.append(tick)
-
-        if self._lastNumMajorTicks < numticks:
-            protoTick = self.majorTicks[0]
-            for i in range(self._lastNumMajorTicks, len(self.majorTicks)):
-                tick = self.majorTicks[i]
-                if self._gridOnMajor:
-                    tick.gridOn = True
-                self._copy_tick_props(protoTick, tick)
 
-        self._lastNumMajorTicks = numticks
-        ticks = self.majorTicks[:numticks]
+        while len(self.majorTicks) < numticks:
```

Comment:
Did we decide that the `_lastNumMajorTicks` logic wasn't needed?  What did all that do and why can we remove it?  Certainly your change is simpler, but it seems most of the un-simpleness in the old code was because of the `_lastNumMajorTicks` logic.

Reply:
It's private, unused anywhere else, and, to the best of my understanding, this implementation does the same thing...